R/format_AMM.R

```diff
@@ -0,0 +1,532 @@
+    dplyr::left_join(Sex_genetic, by = "IndvID") %>%
+    dplyr::left_join(Brood_swap_info, by = "IndvID") %>%
+    dplyr::select(.data$IndvID, .data$Species, .data$PopID,
+                  .data$BroodIDLaid, .data$BroodIDFledged,
+                  .data$RingSeason, .data$RingAge,
+                  .data$Sex_calculated, .data$Sex_genetic)
+
+
+}
+
+#' Create location data table for Ammersee, Germany.
+#'
+#' Create location data table in standard format for data from Ammersee, Germany.
+#' @param data Data frame. Primary data from Ammersee.
+#'
+#' @return A data frame.
+
+create_location_AMM <- function(Capture_data, connection) {
+
+  Habitat_data <- dplyr::tbl(connection, "HabitatDescription") %>%
+    dplyr::select(NestBox, Beech:OtherTree) %>%
+    dplyr::collect() %>%
+    dplyr::group_by(NestBox) %>%
+    dplyr::summarise(across(everything(), ~sum(.x, na.rm = TRUE)), .groups = "keep") %>%
+    dplyr::summarise(DEC = sum(c(Beech, Larch, Maple, Birch, Oak, Willow, Poplar, Alder, AshTree)),
+                     EVE = sum(c(Spruce, Pine))) %>%
+    dplyr::mutate(perc_dec = DEC/(DEC + EVE),
+                  dominant_sp = dplyr::case_when(perc_dec >= 0.66 ~ "DEC",
+                                                 perc_dec < 0.33 ~ "EVE",
+                                                 TRUE ~ "MIX"))
+
+  #The vast majority of nestboxes (90%) of nestboxes are surrounded by deciduous or mixed stands. Therefore,
+  #we use the 'DEC' category to describe the population.
+  #In the future, we could include a nest-box specific habitat type, but that would require us to decide
+  #the range over which habitat is sampled (the nestbox vicinity, the plot?). To do later.
+  table(Habitat_data$dominant_sp)
+
+  start_year <- min(Capture_data$BreedingSeason)
+
+  Location_data <- dplyr::tbl(connection, "NestBoxes") %>%
+    dplyr::collect() %>%
+    dplyr::filter(.data$NestBox != -99L) %>%
+    dplyr::mutate(LocationID = as.character(.data$NestBox),
+                  NestboxID = as.character(.data$NestBox),
+                  Latitude = as.numeric(.data$CoordinateLatitude2013), ## Needed because these variables are stored as
+                  Longitude = as.numeric(.data$CoordinateLongitude2013), ## named vectors, not regular numeric vectors
+                  LocationType = "NB",
+                  PopID = "AMM",
+                  StartSeason = start_year,
+                  EndSeason = dplyr::case_when(nchar(.data$NestboxID) == 4 & stringr::str_sub(.data$NestboxID, 1, 2) == "16" ~ 2016L,
+                                               TRUE ~ 2019L),
+                  HabitatType = "DEC") %>%
+    dplyr::select(.data$LocationID,
+                  .data$NestboxID,
+                  .data$LocationType,
+                  .data$PopID,
+                  .data$Latitude,
+                  .data$Longitude,
+                  .data$StartSeason,
+                  .data$EndSeason,
+                  .data$Habitat)
```

Comment:
After checking out origin/AMM_pipeline, I ran `run_pipelines(PopID = "AMM", save = FALSE)`, which resulted in an error related to these lines: `Error: Column 'Habitat' not found in '.data'`